src/map.cpp

```diff
@@ -4729,11 +4729,11 @@ item &map::add_item_at( const tripoint &p,
 item map::water_from(const tripoint &p)
 {
     if( has_flag( "SALT_WATER", p ) ) {
-        item ret( "salt_water", 0 );
+        item ret( "salt_water", 0, std::numeric_limits<long>::max() );
         return ret;
     }
 
-    item ret( "water", 0 );
+    item ret( "water", 0, std::numeric_limits<long>::max() );
```

Comment:
Actually, I propose changing `charges` back to `int`.

`long` and `int` are the same on many platforms (including my pretty normal Linux 32bit), so it doesn't really give any advantages. It causes other problems:

Out of curiosity, I annotated some of the printf-like functions with `__attribute__((format (printf, 1, 2)))`, which makes gcc check the supplied format string. It issues warnings if the contained format specifiers don't match the arguments.

I get several of those warnings:
```
src/iexamine.cpp:456:53: warning: format '%d' expects argument of type 'int', but argument 2 has type 'long int' [-Wformat=]
```
They are caused by code like this: `string_format( "Money left: %d", card->charges )`. Here "%d" is used to print the charges, it requires an `int`, but charges is a `long`.